File: code/library/src/main/java/org/fbme/lib/iec61499/parser/ServiceInterfaceFBTypeConverter.java
package org.fbme.lib.iec61499.parser;


import org.fbme.lib.common.Identifier;
import org.fbme.lib.iec61499.declarations.ServiceInterfaceFBTypeDeclaration;
import org.jetbrains.annotations.Nullable;

public class ServiceInterfaceFBTypeConverter extends DeclarationConverterBase<ServiceInterfaceFBTypeDeclaration> {

    public ServiceInterfaceFBTypeConverter(ConverterArguments arguments) {
        super(arguments);
    }

    @Override
    protected ServiceInterfaceFBTypeDeclaration extractDeclarationBody(@Nullable Identifier identifier) {
        ServiceInterfaceFBTypeDeclaration std = myFactory.createServiceInterfaceFBTypeDeclaration(identifier);
        new FBInterfaceConverter(this, std).extractInterface();
        return std;
    }
}

File: code/library/src/main/kotlin/org/fbme/lib/iec61499/parser/ServiceInterfaceFBTypeConverter.kt
package org.fbme.lib.iec61499.parser

import org.fbme.lib.common.Identifier
import org.fbme.lib.iec61499.declarations.ServiceInterfaceFBTypeDeclaration

class ServiceInterfaceFBTypeConverter(arguments: ConverterArguments) :
    DeclarationConverterBase<ServiceInterfaceFBTypeDeclaration>(arguments) {
    override fun extractDeclarationBody(identifier: Identifier?): ServiceInterfaceFBTypeDeclaration {
        val std = factory.createServiceInterfaceFBTypeDeclaration(identifier)
        FBInterfaceConverter(this, std).extractInterface()
        return std
    }
}
